### PR TITLE
Allows sending custom values in the geocoder analysis

### PR DESCRIPTION
**Pull request by javierarce**
This PR updates the geocoder analyses to support sending columns and custom values to the geocoder. 

For example, it'll be possible to manually specify the name of a city in the Street Address analysis instead of selecting a column.

CR: @xavijam

### CODE REVIEW

File: lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/georeference-admin-region.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+var _ = require('underscore');
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/georeference-admin-region.js]: georeference-admin-region-model.js ? 🙏

File: lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/georeference-city.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
+var _ = require('underscore');
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/georeference-city.js]: ...-model please :)

File: lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/georeference-city.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  },
+
+  validatorFor: {
+    city_column: ['required']
+  },
+
+  parametersDataFields: 'city_column,admin_region,country',
+
+  parse: function (nodeAttrs) {
+    var opts = {
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/georeference-city.js]: Why not return directly this object with the type, it is static so...

File: lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/georeference-admin-region.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    }
+  },
+
+  validatorFor: {
+    admin_region_column: ['required']
+  },
+
+  parametersDataFields: 'admin_region_column,country',
+
+  parse: function (nodeAttrs) {
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/georeference-admin-region.js]: add the proper type directly to this object and return it?

File: lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/georeference-city.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+      showSearch: true,
+      allowFreeTextInput: true
+    }
+  },
+
+  validatorFor: {
+    city_column: ['required']
+  },
+
+  parametersDataFields: 'city_column,admin_region,country',
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/georeference-city.js]: I like those things in an array, better to manipulate them later, but super minor.

File: lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/georeference-country.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+  },
+
+  validatorFor: {
+    country_column: ['required']
+  },
+
+  parametersDataFields: 'country_column',
+
+  parse: function (nodeAttrs) {
+    var opts = {
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/georeference-country.js]: Dixit! Brexit!

File: lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/georeference-ip-address.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
+var _ = require('underscore');
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/georeference-ip-address.js]: -model :)

File: lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/georeference-long-lat.js [hunk 8]
```diff
@@ -0,0 +1,30 @@
+module.exports = {
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/georeference-long-lat.js]: -model :) 💃

File: lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/georeference-long-lat.js [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+      showSearch: true
+    }
+  },
+
+  validatorFor: {
+    latitudde: ['required'],
+    longitude: ['required']
+  },
+  parse: function (nodeAttrs) {
+    return {
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/georeference-long-lat.js]: Hey, you returning type directly here! :)

File: lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/georeference-postal-code.js [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+  },
+
+  validatorFor: {
+    city_column: ['required']
+  },
+
+  parametersDataFields: 'postal_code_column,country',
+
+  parse: function (nodeAttrs) {
+    var opts = {
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/georeference-postal-code.js]: Dixit!

File: lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/georeference-postal-code.js [hunk 11]
```diff
@@ -0,0 +1,42 @@
+var _ = require('underscore');
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/georeference-postal-code.js]: -model :)

File: lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/georeference-street-address.js [hunk 12]
```diff
@@ -0,0 +1,55 @@
+var _ = require('underscore');
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/georeference-street-address.js]: -model :)

File: lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/georeference-types.js [hunk 13]
```diff
@@ -0,0 +1,9 @@
+module.exports = [
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/georeference-types.js]: uhhhh, if we change the model thing in all files, let's take care of these references.

**Comment by xavijam** [comment]
:+1:, just those comments, I feel they are minor.

**Comment by saleiva** [comment]
VAMOS NENOS!

**Comment by javierarce** [comment]
It seems there are some issues in the staging servers with the analyses. Waiting until I can fully test this PR there to merge it.

**Comment by javierarce** [comment]
Ok, once it's green I'll merge this. Sorry for the delay!

**Comment by javierarce** [comment]
Fixed an error introduced in this PR: https://github.com/CartoDB/cartodb/pull/8995. The other  failing spec will be fixed here https://github.com/CartoDB/cartodb/pull/8993.

**Comment by javierarce** [comment]
Talked to @alonsogarciapablo and @xavijam about the failing spec. It'll will be fixed in an upcoming PR.

**Comment by xavijam** [comment]
Merge, frontend tests passed 💃

**Pull request merged**